spec/requests/brands_controller_spec.rb

```diff
@@ -0,0 +1,258 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe BrandsController, type: :request do
+  let(:manager) { create(:account, role: create(:role)) }
+  let(:staff) { create(:account, role: create(:role, name: 'Staff')) }
+
+  before(:each) do
+    @brand1 = create(:brand)
+    @brand2 = create(:brand, name: 'Samsung')
+    create(:model, brand: @brand1)
+    create(:model, name: 'iPhone 7', brand: @brand1)
+    create(:model, name: 'Galaxy Note 10', brand: @brand2)
+  end
+
+  # brands#index
+  describe 'GET /brands' do
+    shared_examples 'render index successfully' do
+      it 'render index page' do
+        get brands_path
+        expect(response).to render_template(:index)
+      end
+
+      it 'render all brands' do
+        get brands_path
+        expect(assigns(:brands).count).to eq(2)
+      end
+    end
+
+    context 'when user logged in as manager' do
+      before(:each) do
+        sign_in manager
+      end
+
+      include_examples 'render index successfully'
+    end
+
+    context 'when user logged in as staff' do
+      before(:each) do
+        sign_in staff
+      end
+
+      include_examples 'render index successfully'
+    end
+
+    context 'when user logged out' do
+      it 'redirect to log-in page' do
+        get brands_path
+        expect(response).to redirect_to(new_account_session_path)
+      end
+    end
+  end
+
+  # brands#create
+  describe 'POST /brands/create' do
+    context 'when user logged in as manager' do
+      before(:each) do
+        sign_in manager
+      end
+
+      context 'with valid attributes' do
+        before(:each) do
+          post brands_create_path, params: { name: 'A new brand' }
+        end
+
+        it 'created successfully' do
+          expect(Brand.count).to eq(3)
```

Comment:
FYI, Ngoài cách trên, có thể dùng cách này dễ đọc hơn
expect { subject }.to change  { Brand.count }.from(2).to(3)